### PR TITLE
signal: catch all termination signals by default

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we modify the primary `signal` package to instead catch
all signals. Before this commit, it would only catch the interrupt
signal sent from the kernel. With this new commit, we'll now also catch
(or attempt to catch): `SIGABRT`, `SIGTERM`, `SIGSTOP`, and `SIGQUIT`.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Before this PR, we didn't attempt to catch `SIGTERM` which is customarily used by `systemd` to shutdown active services. Many of the raspberry pi guides circulating out there use a `ssytemd` set up to handle creation/destruction of the `lnd` process.

File: signal/signal.go [hunk 1]
```diff
@@ -26,7 +27,11 @@ var (
 )
 
 func init() {
-	signal.Notify(interruptChannel, os.Interrupt)
+	signalsToCatch := []os.Signal{
+		os.Interrupt, os.Kill, syscall.SIGABRT, syscall.SIGTERM,
+		syscall.SIGSTOP, syscall.SIGQUIT,
```
Comment (cfromknecht) [inline signal/signal.go]: style nit: new line for each signal?

Comment (Roasbeef) [reply signal/signal.go]: Fixed, also we'll now log the signal itself.

**Comment by vamp111** [comment]
expected update reliese anytime soon ?

**Pull request merged**